### PR TITLE
Support 'func init' for new Node.js programming model

### PR DESCRIPTION
**Pull request by ejizba**
### Issue describing the changes in this PR

This is the "P0" requirements for the new Node.js programming model - (primarily "func init" and a few info messages). It should very closely match the existing Python experience. Everything is done and ready to be reviewed, but unfortunately we're not ready to officially announce public preview for a few months so I made it a "draft PR".

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [x] Otherwise: Documentation for new programming model is handled elsewhere
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/Azure.Functions.Cli/Helpers/ProgrammingModelHelper.cs [hunk 1]
```diff
@@ -15,12 +15,15 @@ public static IEnumerable<ProgrammingModel> GetProgrammingModels()
...
         {
-            var allProgrammingModels = GetProgrammingModels();
-            if (workerRuntime != WorkerRuntime.python)
+            switch (workerRuntime)
             {
-                return allProgrammingModels.Where(pm => pm != ProgrammingModel.V2);
+                case WorkerRuntime.python:
+                    return new List<ProgrammingModel>() { ProgrammingModel.V1, ProgrammingModel.V2 };
+                case WorkerRuntime.node:
+                    return new List<ProgrammingModel>() { ProgrammingModel.V3, ProgrammingModel.V4 };
```
Comment (khkh-ms) [inline src/Azure.Functions.Cli/Helpers/ProgrammingModelHelper.cs:23]: So we don't have v1 and v2 in Node? Looks like we have gone through multiple models for node.

**Review by khkh-ms — APPROVED** [review]
Looks good. Have only one question.

**Comment by khkh-ms** [comment]
Requires integration test fix.

File: src/Azure.Functions.Cli/Helpers/ProgrammingModelHelper.cs (hunk 1 above)
Comment (ejizba) [reply src/Azure.Functions.Cli/Helpers/ProgrammingModelHelper.cs:23]: v1 and v2 of the node programming model were associated with host runtime v2 and v3 which are out of support by now

**Comment by kshyju** [comment]
@ejizba Is this change still valid/relevant? If so, can you please rebase and fix the conflicts?

**Comment by ejizba** [comment]
Nope, no longer relevant, thanks

**Pull request closed without merging**